File: prime_factor.py
from math import sqrt


def prime_factors(num):
    factors = []
    isPrime = 1
    n = num
    while n % 2 == 0:
        factors.append(2)
        n //= 2
        isPrime = 0
    for i in range(3, int(sqrt(n)) + 1):
        while n % i == 0:
            isPrime = 0
            factors.append(i)
            n //= i
    if (n > 2) & isPrime:
        print(num, "is prime number")
    else:
        factors.append(n)
        print(factors)


num = 987654321
prime_factors(num)

File: prime-number/prime_factor.c
#include <stdio.h>
#include <math.h>

int prime_factor(unsigned long long num)
{
    unsigned long long n = num, factor;
    const int len = 100;
    unsigned long long factors[len];
    int x = 0;
    // even number
    while (n % 2 == 0)
    {
        factors[x] = 2;
        x++;
        n /= 2;
    }

    for (int i = 3; i <= sqrt(n); i += 2)
    {
        while (n % i == 0)
        {
            factors[x] = i;
            x++;
            n = n / i;
        }
    }

    if ((n > 2) & (x == 0))
    {
        printf("%llu is a prime number\n", n);
    }
    else
    {
        factors[x] = n;
        x++;
    }

    printf("the prime factor of %llu is ", num);
    for (int i = 0; i <= x - 1; i++)
    {
        printf("%llu ", factors[i]);
    }
}

int lpf(unsigned long long n)
{
    unsigned long long factor;
    for (int i = 2; i <= n; i++)
    {
        if (n % i == 0)
        {
            factor = n;
            n = n / i;
            if (n == 1)
            {
                printf("%d is the largest prime factor.\n", factor);
                break;
            }
        }
    }
}

int main()
{
    unsigned long long num1 = 14695981039346656037ULL;
    prime_factor(num1);
}